ambry-network/src/main/java/com/github/ambry/network/NettyServerRequestResponseChannel.java

```diff
@@ -37,6 +36,7 @@ public class NettyServerRequestResponseChannel implements RequestResponseChannel
   private final NetworkRequestQueue networkRequestQueue;
   private final ServerMetrics serverMetrics;
   private final ServerRequestResponseHelper requestResponseHelper;
+  protected static final Logger publicAccessLogger = LoggerFactory.getLogger("PublicAccessLogger");
```

Comment:
I'm not sure how the LoggerFactory works. With that, will publicAccessLogger write to the public log instead of the ambry log?

Also, right now we are using LogManager instead of LoggerFactory for the PublicAccessLogger. I don't know the difference between them.
  private static final Logger publicAccessLogger = LogManager.getLogger("PublicAccessLogger");

Reply:
Yeah, I think the Logger obtained via `PublicAccessLogger` name will write logs to public access logs. I think log4j2 has concept of "appenders" which allows us to choose the destination log file for a logger via configuration. If we see in internal AmbryLI configuration, the appender for `PublicAccessLogger` redirects logs to public access logs.

LogManager is used for structured logging. But I think we only use it for frontends. For servers, we are still using LoggerFactory. All the server public access logging happens in AmbryRequests.java. 

This is my understanding :)